Contribute/dotnet-style-guide.md

```diff
@@ -0,0 +1,263 @@
+
+To link to a generic type, such as [System.Collections.Generic.List\<T>](https://docs.microsoft.com/dotnet/api/system.collections.generic.list-1), you use the ` (%60) character followed by the number of generic type parameters. For example, \<xref:System.Nullable%601> links to the [System.Nullable\<T>](https://docs.microsoft.com/dotnet/api/system.nullable-1) type, while \<xref:System.Func%602> links to the [System.Func\<T,TResult>](https://docs.microsoft.com/dotnet/api/system.func-2) delegate.
+
+## Code
+
+The best way to include code is to include snippets from a working sample. Create your sample following the instructions in the [contributing to .NET](dotnet-contribute.md#contributing-to-samples) article. The basic rules for including code are located in the general guidance on [code](how-to-write-use-markdown.md#code-includes).
+
+You can include the code using the following syntax:
+
+```markdown
+[!code-<language>[<name>](<pathToFile><queryoption><queryoptionvalue>)]
+```
+
+* `-<language>` (*optional* but *recommended*)
+  * Language of the code snippet being referenced. For a list of supported values, see [Supported languages](#supported-languages).
+
+* `<name>` (*optional*)
+  * Name for the code snippet. It doesn’t have any impact on the output HTML, but you can use it to improve the readability of your Markdown source.
+
+* `<pathToFile>` (*mandatory*)
+  * Relative path in the file system that indicates the code snippet file to reference.
+
+* `<queryoption>` (*optional*)
+  * Used to specify how the code should be retrieved from the file:
+    * `#`:  `#L{startlinenumber}-L{endlinenumber}` (line range) *or* `#{tagname}` (tag name).
+    We discourage the use of line numbers because they are very brittle. Tag name is the preferred way of referencing code snippets.
+    * `range`: `?range=1,3-5` A range of lines. This example includes lines 1, 3, 4, and 5.
+
+We recommend using the tag name option whenever possible. The tag name is the name of a region or of a code comment in the format of `Snippettagname` present in the source code. The following example shows how to refer to the tag name `BasicThrow`:
+
+```markdown
+[!code-csharp[csrefKeyword#1](../../../../samples/snippets/csharp/language-reference/keywords/throw/throw-1.cs#BasicThrow)]
+```
+
+And you can see how the snippet tags are structured in [this source file](https://github.com/dotnet/samples/blob/master/snippets/csharp/language-reference/keywords/throw/throw-1.cs). For details about tag name representation in code snippet source files by language, see the [DocFX guidelines](https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#tag-name-representation-in-code-snippet-source-file).
```

Comment:
As mentioned earlier, the example file might be not the best example of the snippet structure.